src/adapter/src/coord/sequencer/inner/cluster.rs

```diff
@@ -252,23 +275,41 @@ impl Coordinator {
                             .expect("There must be an active connection")
                             .pending_cluster_alters
                             .insert(cluster_id.clone());
-                        let span = Span::current();
                         let new_config_managed = new_config_managed.clone();
-                        Ok(StageResult::Handle(mz_ore::task::spawn(
-                            || "Finalize Alter Cluster",
-                            async move {
-                                tokio::time::sleep(duration).await;
-                                let stage = ClusterStage::Finalize(FinalizeAlterCluster {
-                                    validity,
-                                    plan,
-                                    new_config: new_config_managed,
-                                });
-                                Ok(Box::new(stage))
+                        match &strategy {
+                            AlterClusterPlanStrategy::None => coord_bail!("AlterClusterPlanStrategy must not be None if NeedsFinalization is Yes"),
```

Comment:
This returns an `AdapterError::Unstructured`, but we really should be returning an `AdapterError::Internal`.